Automattic/jetpack
# Introduce Jetpack Reach plugin

Introduces a plugin named Jetpack Reach with the ability to connect the site and show My Jetpack.

<img width="1769" alt="image" src="https://user-images.githubusercontent.com/746152/156755432-96f478e9-6845-4327-8d1b-bd59005319a8.png">
<img width="1766" alt="image" src="https://user-images.githubusercontent.com/746152/156755566-a2e3abb5-64e8-4ba0-8a92-452a3dbc4d9b.png">
#### Changes proposed in this Pull Request:


Created with the `jetpack generate` tool but also this was added:
* Loaded autoloader 
* Added basic internal monorepo dependencies
  * Added My Jetpack as dependency
  * Added Jetpack autoloader as dependency
  * Added Jetpack\Admin_UI as dependency
  * Added Jetpack\Assets as dependency
  * Added composer plugin to dependencies
* Added PHPCS config file
* Renamed `example.php` to `class-jetpack-reach.php`
* Renamed `reach.php to jetpack-reach.php`
* Added WordBless `wordpress` dir ,  `build` and `.cache` to .gitignore
* Introduced Admin Page structure
  * Added `.eslintrc.js`
    * Added `babel.config.js`
  * Added `webpack.config.js`
  * Copied `package.json` from Backup's
  * Added `src/js/index.js` and `components/admin-page`.
* Added `build-development`, `build-production` and `watch` scripts to compose
* Added `Jetpack\Config` as dependency.
  *  [Added Jetpack\Identity_Crisis and Jetpack\Sync as dependency] (https://github.com/Automattic/jetpack/pull/23232/commits/b7ee88a3ed86443a2c894bc89cabf9b2669dbaf1) (Had to add them as they're depedency of Config.
* Added ability to connect and connected/disconnected screens on admin page.
* Updated `.gitattributes` with similar values to those on Backup's. The vendor and jetpack_vendor dirs were absent from the action builds.
* Added autoloader-suffix to composer.json (9e39e5dcd4382e4ced053b797b6bafdef54e2f31)
  * Achieved it by running composer config autoloader-suffix c4802e05bbcf59fd3b6350e8d3e5482c_reachⓥ0.1.0-alpha as recommended by the log of Lint/Project structure check.

#### Jetpack product discussion

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:

1. Launch a [JN site with this branch and plugin](https://jurassic.ninja/create/?jetpack-beta&nojetpack&branches.jetpack-reach=add/reach-plugin&wp-debug-log) (This link launches without the main Jetpack plugin).
3. Visit the Jetpack reach menu
  
<img width="560" alt="image" src="https://user-images.githubusercontent.com/746152/157232961-d07a5a23-4d00-4dcb-aae1-356adba2a132.png">
2. Connect the site.
3. Expect to se the My Jetpack menu item
